os/glog/glog_logger_rotate.go

```diff
@@ -137,6 +138,11 @@ func (l *Logger) rotateChecksTimely(ctx context.Context) {
 		intlog.Errorf(ctx, `%+v`, err)
 	}
 	intlog.Printf(ctx, "logging rotation start checks: %+v", files)
+	// get file name regex pattern
+	// access-{y-m-d}-test.log => access-$-test.log => access-\$-test\.log => access-(.+?)-test\.log
+	fileNameRegexPattern, _ := gregex.ReplaceString(`{.+?}`, "$", l.config.File)
+	fileNameRegexPattern = gregex.Quote(fileNameRegexPattern)
+	fileNameRegexPattern = strings.ReplaceAll(fileNameRegexPattern, "\\$", "(.+?)")
```

Comment:
`strings.ReplaceAll` or `gregex.ReplaceString`?

Reply:
`fileNameRegexPattern, _ := gregex.ReplaceString(`{.+?}`, "$", l.config.File)`   
using regex to replace "access-{y-m-d}-test.log" to "access-$-test.log".  The dollar symbol ($) is only a placeholder.
`fileNameRegexPattern = gregex.Quote(fileNameRegexPattern)` 
using regex quote to replace "access-$-test.log" to "access-\$-test\.log". After this replacement, the "fileNameRegexPattern" is only a normal string, not regex.
`fileNameRegexPattern = strings.ReplaceAll(fileNameRegexPattern, "\\$", "(.+?)")` 
using string replacement to change "access-\$-test\.log" to "access-(.+?)-test\.log", now, "fileNameRegexPattern" become real regex.

above commit changes are for one purpose: change gf's date format {y-m-d} to regex (.+?).